akka-sample-sharding-scala/killrweather-fog/src/main/scala/sample/killrweather/fog/Fog.scala

```diff
@@ -0,0 +1,94 @@
+package sample.killrweather.fog
+
+import java.net.{DatagramSocket, InetSocketAddress}
+import java.nio.channels.DatagramChannel
+
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ActorSystem, Behavior}
+import com.typesafe.config.{Config, ConfigFactory}
+
+/**
+  * In another terminal start the `Fog` (see Fog computing https://en.wikipedia.org/wiki/Fog_computing).
+  * Starts the fog network, simulating devices and stations.
+  * In the wild, each station would run its own system and be location-aware.
+  */
+object Fog {
+
+  def main(args: Array[String]): Unit = {
+
+    val ports = if (args.isEmpty) Seq(8081) else args.map(_.toInt).toSeq
+
+    ActorSystem[Nothing](Guardian(ports.toVector), "Fog")
+  }
+}
+
+object Guardian {
+
+  def apply(ports: Vector[Int]): Behavior[Nothing] = {
+    Behaviors.setup[Nothing] { context =>
+      val settings = FogSettings(context.system)
+      val weatherPorts = ports.flatMap(settings.findHttpPort)
+
+      (1 until settings.weatherStations).foreach { n =>
+        val wsid = WeatherStation.WmoId(n.toString)
+        val weatherPort = weatherPorts(n % weatherPorts.size)
+
+        context.spawn(WeatherStation(wsid, settings, weatherPort), s"weather-station-${wsid.id}")
+      }
+      Behaviors.empty
+    }
+  }
+}
+
+object FogSettings {
+
+  def apply(system: ActorSystem[_]): FogSettings = {
+    apply(system.settings.config.getConfig("killrweather"))
+  }
+
+  def apply(config: Config): FogSettings = {
+    import akka.util.Helpers.Requiring
+
+    val config =
+      ConfigFactory.load("application.conf").getConfig("killrweather")
```

Comment:
strange to load the config again, when it's already passed in as parameter

Reply:
Oops that's a line I forgot to finish refactoring, fixing.